independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/decorators/decorated/InvalidDecoratedTest.java

```diff
@@ -0,0 +1,40 @@
+package io.quarkus.arc.test.decorators.decorated;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.Decorated;
+import jakarta.enterprise.inject.spi.Bean;
+import jakarta.inject.Inject;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.arc.test.ArcTestContainer;
+
+public class InvalidDecoratedTest {
+
+    @RegisterExtension
+    public ArcTestContainer container = ArcTestContainer.builder()
+            .beanClasses(InvalidBean.class)
+            .shouldFail()
+            .build();
+
+    @Test
+    public void testDecoration() {
+        assertNotNull(container.getFailure());
+        assertTrue(container.getFailure().getMessage().startsWith(
+                "Invalid injection of @Decorated Bean<T>, can only be injected into decorators but was detected in: "
+                        + InvalidBean.class.getName() + "#decorated"));
+    }
+
+    @ApplicationScoped
+    static class InvalidBean {
+
+        @Inject
+        @Decorated
+        Bean<?> decorated;
```

Comment:
This is actually not a valid injection point - there has to be a type corresponding to the delegate type, [see this section of CDI spec](https://jakarta.ee/specifications/cdi/4.0/jakarta-cdi-spec-4.0#bean_metadata_full).
This means some form of validation is missing.

Reply:
Ah, I didn't notice Ladislav beat me to it with his comment - https://github.com/quarkusio/quarkus/pull/43864#issuecomment-2413148607
Just apply his changes to add the validation :)

Reply:
just merged it ! thanks a lot !